moto/codedeploy/models.py

```diff
@@ -0,0 +1,496 @@
+"""CodeDeployBackend class with methods for supported APIs."""
+
+import uuid
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from moto.core.base_backend import BackendDict, BaseBackend
+from moto.core.common_models import BaseModel
+from moto.core.utils import iso_8601_datetime_with_milliseconds
+
+from .exceptions import (
+    ApplicationAlreadyExistsException,
+    ApplicationDoesNotExistException,
+    ApplicationNameRequiredException,
+    DeploymentDoesNotExistException,
+    DeploymentGroupAlreadyExistsException,
+    DeploymentGroupDoesNotExistException,
+    DeploymentGroupNameRequiredException,
+)
+
+
+class Application(BaseModel):
+    def __init__(
+        self, application_name: str, compute_platform: str, tags: List[Dict[str, str]]
+    ):
+        self.id = str(uuid.uuid4())
+        self.application_name = application_name
+        self.compute_platform = compute_platform
+        self.tags = tags.copy() if tags else []
+
+        # Boto docs mention that the field should be datetime, but AWS API says number
+        self.create_time = iso_8601_datetime_with_milliseconds()
+
+        # these GitHub fields need to be set by the user in the console
```

Comment:
Great comments throughout this PR! They give some very useful context for people not familiar with the service